index.js

```diff
@@ -37,7 +37,9 @@ const mergedObjects = data => data.reduce((acc, item) => {
 const getGeneIdsFromGeneRule = (geneRule) => {
   let idList = [];
   if (geneRule) {
-    idList = geneRule.split(/\s+and\s+|\s+or\s+/).filter(e => e);
+    idList = geneRule.split(/\s+and\s+|\s+or\s+/)
```

Comment:
Good catch @e0 ! There are some more complicated Gene rules such as for `ATPasel`, in which multiple layers of parentheses exists. Current code can not handle such cases. 

In addition, it will be great that the original `and or` logic preserves after parsing. 

```
    - !!omap
      - id: "ATPasel"
      - name: "V-Type ATPase, H+ Transporting, Lysosomal"
      - metabolites: !!omap
          - m01285c: 1
          - m01371c: -1
          - m02039l: 1
          - m02040c: -1
          - m02751c: 1
      - lower_bound: 0
      - upper_bound: 1000
      - gene_reaction_rule: "((ENSG00000105929 and ENSG00000113732 and ENSG00000117410 and ENSG00000147614 and ENSG00000185883) or (ENSG00000113732 and ENSG00000117410 and ENSG00000147614 and ENSG00000185344 and ENSG00000185883) or (ENSG00000105929 and ENSG00000113732 and ENSG00000117410 and ENSG00000159720 and ENSG00000185883) or (ENSG00000033627 and ENSG00000113732 and ENSG00000117410 and ENSG00000147614 and ENSG00000185883) or (ENSG00000033627 and ENSG00000113732 and ENSG00000117410 and ENSG00000159720 and ENSG00000185883) or (ENSG00000110719 and ENSG00000113732 and ENSG00000117410 and ENSG00000159720 and ENSG00000185883) or (ENSG00000113732 and ENSG00000117410 and ENSG00000159720 and ENSG00000185344 and ENSG00000185883) or (ENSG00000110719 and ENSG00000113732 and ENSG00000117410 and ENSG00000147614 and ENSG00000185883)) and ((ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000128524 and ENSG00000136888 and ENSG00000147416 and ENSG00000155097 and ENSG00000250565) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000116039 and ENSG00000128524 and ENSG00000136888 and ENSG00000155097 and ENSG00000250565) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000128524 and ENSG00000136888 and ENSG00000143882 and ENSG00000147416 and ENSG00000250565) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000128524 and ENSG00000131100 and ENSG00000136888 and ENSG00000147416 and ENSG00000155097) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000116039 and ENSG00000128524 and ENSG00000131100 and ENSG00000136888 and ENSG00000155097) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000128524 and ENSG00000147416 and ENSG00000155097 and ENSG00000213760 and ENSG00000250565) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000128524 and ENSG00000143882 and ENSG00000147416 and ENSG00000213760 and ENSG00000250565) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000128524 and ENSG00000131100 and ENSG00000147416 and ENSG00000155097 and ENSG00000213760) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000116039 and ENSG00000128524 and ENSG00000131100 and ENSG00000151418 and ENSG00000155097) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000128524 and ENSG00000143882 and ENSG00000147416 and ENSG00000151418 and ENSG00000250565) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000116039 and ENSG00000128524 and ENSG00000131100 and ENSG00000143882 and ENSG00000213760) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000116039 and ENSG00000128524 and ENSG00000143882 and ENSG00000213760 and ENSG00000250565) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000116039 and ENSG00000128524 and ENSG00000131100 and ENSG00000155097 and ENSG00000213760) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000128524 and ENSG00000147416 and ENSG00000151418 and ENSG00000155097 and ENSG00000250565) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000128524 and ENSG00000131100 and ENSG00000147416 and ENSG00000151418 and ENSG00000155097) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000116039 and ENSG00000128524 and ENSG00000155097 and ENSG00000213760 and ENSG00000250565) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000116039 and ENSG00000128524 and ENSG00000151418 and ENSG00000155097 and ENSG00000250565) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000128524 and ENSG00000131100 and ENSG00000143882 and ENSG00000147416 and ENSG00000151418) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000128524 and ENSG00000131100 and ENSG00000143882 and ENSG00000147416 and ENSG00000213760) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000116039 and ENSG00000128524 and ENSG00000131100 and ENSG00000143882 and ENSG00000151418) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000116039 and ENSG00000128524 and ENSG00000143882 and ENSG00000151418 and ENSG00000250565) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000116039 and ENSG00000128524 and ENSG00000136888 and ENSG00000143882 and ENSG00000250565) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000116039 and ENSG00000128524 and ENSG00000131100 and ENSG00000136888 and ENSG00000143882) or (ENSG00000047249 and ENSG00000100554 and ENSG00000114573 and ENSG00000128524 and ENSG00000131100 and ENSG00000136888 and ENSG00000143882 and ENSG00000147416))"
```

Reply:
It looks like the regex needs some revision. As @nanjiangshu correctly noted, we must not alter the logic in any way.

Reply:
The parenthesis nesting has been resolved in [this commit](https://github.com/MetabolicAtlas/neo4j-data-generation/pull/18/commits/c23570dadacd9341804a47630a8b9a15109ce859).